src/main/java/com/coveros/selenified/application/Check.java

```diff
@@ -0,0 +1,815 @@
+    }
+
+    /**
+     * Checks that a cookie doesn't exist in the application with the provided
+     * cookieName. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param unexpectedCookieName - the name of the cookie
+     */
+    void cookieNotExists(String unexpectedCookieName);
+
+    /**
+     * Checks that a cookie doesn't exist in the application with the provided
+     * cookieName. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support. Handles the actual
+     * logging for the verify, assert, and waitFor implementations
+     *
+     * @param unexpectedCookieName - the name of the cookie
+     * @param waitFor              - if waiting, how long to wait for (set to 0 if no wait is desired)
+     * @param timeTook             - the amount of time it took for wait for something (assuming we had to wait)
+     * @return Boolean: whether the cookie is present or not
+     */
+    default boolean checkCookieNotExists(String unexpectedCookieName, double waitFor, double timeTook) {
+        //record the action
+        getOutputFile().recordAction("to find no cookie with the name <b>" + unexpectedCookieName + STORED, waitFor);
+        // check for the object to be present
+        boolean isCookiePresent = getApp().is().cookiePresent(unexpectedCookieName);
+        if (isCookiePresent) {
+            getOutputFile().recordActual(COOKIE + unexpectedCookieName + STORED, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(COOKIE + unexpectedCookieName + NOT_STORED, timeTook, Success.PASS);
+        }
+        return !isCookiePresent;
+    }
+
+    /**
+     * Checks that a cookies with the provided name has a value equal to the
+     * expected value. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param cookieName          - the name of the cookie
+     * @param expectedCookieValue - the expected value of the cookie
+     */
+    void cookieEquals(String cookieName, String expectedCookieValue);
+
+    /**
+     * Checks that a cookies with the provided name has a value equal to the
+     * expected value. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support. Handles the actual
+     * logging for the verify, assert, and waitFor implementations
+     *
+     * @param cookieName          - the name of the cookie
+     * @param expectedCookieValue - the expected value of the cookie
+     * @param waitFor             - if waiting, how long to wait for (set to 0 if no wait is desired)
+     * @param timeTook            - the amount of time it took for wait for something (assuming we had to wait)
+     * @return String: if the cookie is present, the value of the cookie, and null if the cookie isn't present
```

Comment:
Shouldn't this return true and false? Another method can be called "getCookieValue" or something. That method can call the true/false version.

Reply:
So, maybe this is poorly named, but it's a helper method, for the actual check implementation, which needs the value to do the final assert, verify or wait

Reply:
And it would read something like app.assert/verify().cookieEquals(), which would make the function more clear.